lib/manager/gradle-wrapper/artifacts.ts

```diff
@@ -0,0 +1,83 @@
+import Git from 'simple-git/promise';
+import { resolve } from 'path';
+import * as fs from 'fs-extra';
+import { logger } from '../../logger';
+import { UpdateArtifact, UpdateArtifactsResult } from '../common';
+import { exec, ExecOptions } from '../../util/exec';
+import { readLocalFile } from '../../util/fs';
+import { platform } from '../../platform';
+import { VERSION_REGEX } from './search';
+import { gradleWrapperFileName, prepareGradleCommand } from '../gradle/index';
+
+async function addIfUpdated(
+  status: Git.StatusResult,
+  filePath: string
+): Promise<UpdateArtifactsResult | null> {
+  if (status.modified.includes(filePath)) {
+    return {
+      artifactError: null,
+      file: {
+        name: filePath,
+        contents: await readLocalFile(filePath),
+      },
+    };
+  }
+  return null;
+}
+
+export async function updateArtifacts({
+  packageFileName,
+  updatedDeps,
+  newPackageFileContent,
+  config,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  try {
+    const projectDir = config.localDir;
+    logger.debug(updatedDeps, 'gradle-wrapper.updateArtifacts()');
+    const gradlewPath = resolve(
+      projectDir,
+      `./${gradleWrapperFileName(config)}`
+    );
+    const version = VERSION_REGEX.exec(newPackageFileContent).groups.version;
+    await prepareGradleCommand(
+      gradleWrapperFileName(config),
+      projectDir,
+      await fs.stat(gradlewPath).catch(() => null),
+      null
+    );
+    const cmd = `${gradlewPath} wrapper --gradle-version ${version} --project-dir ${projectDir}`;
```

Comment:
Have you verified that this behaves as expected if the gradle wrapper file has already been updated to the latest version? i.e. if the `updateDependency()` function updates the version already and then this wrapper command is found, does it continue to install/update the additional files, or does it abort because it thinks the file has already been updated?

Reply:
It works correctly. I have verified that with [tests](https://github.com/renovatebot/renovate/pull/5842/files#diff-b92414beacaefed9b1be945bafddf048R34).